src/controllers/project.controller/updateProject/index.js

```diff
@@ -19,13 +21,81 @@ const updateProject = catchAsync(async (req, res) => {
     },
     changes: {},
   };
-  // if (req.body.caption) variables['changes']['caption'] = req.body.caption;
+
   if (req.body.description) variables.changes.description = req.body.description;
   if (req.body.title) variables.changes.title = req.body.title;
   if (req.body.link) variables.changes.link = req.body.link;
+  if (req.body.status !== undefined) variables.changes.status = req.body.status;
+  if (req.body.completed !== undefined) variables.changes.completed = req.body.completed;
 
   const response = await Hasura(updatePost_query, variables);
 
+  if (req.body.roles_required) {
+    const rolesUpdateVariables = {
+      projectId: id,
+      newRoles: req.body.roles_required.map((role) => ({
+        project_id: id,
+        role_name: role.role_name,
+      })),
+    };
+    await Hasura(updateRolesRequired, rolesUpdateVariables);
+  }
+
+  if (req.body.looking_for_mentors !== undefined || req.body.looking_for_members !== undefined) {
+    const projectFlagsUpdateVariables = {
+      projectId: id,
+      lookingForMentors: req.body.looking_for_mentors,
+      lookingForMembers: req.body.looking_for_members,
+    };
+    await Hasura(updateProjectFlags, projectFlagsUpdateVariables);
+  }
+
+  if (req.body.mentions) variables.changes.mentions = req.body.mentions;
```

Comment:
I see you are making use of the same `variables.changes` here, but I think you missed using it in a query...

Reply:
Oh, you are updating mentions, tags and documents separately, so you can remove L53, 54 and 55